source-academy/cadet-frontend
# Mobile Missions Workspace

### Description



- Refactored AssessmentWorkspace.tsx (Missions Workspace) and Assessment.tsx (List of missions page) to functional components
- Introduced the mobile version of the Missions Workspace
- Improved UI of missions/ quests/ paths/ etc. listing pages at `academy/*`

Note:
- React.useEffect is not fired during Jest snapshot testing with shallow render (see [this](https://github.com/enzymejs/enzyme/issues/2086)), thus there is a minor change to the AssessmentWorkspace snapshot as the `showOverlay` state is not updated after being rendered by Jest. It works as per normal on the browser.

### Type of change


### How to test



- See the Mobile Review Guide [here](https://github.com/source-academy/cadet-frontend/wiki/Mobile-Development-Guide#pull-request-review-guide) for instructions on setup (both mobile browser and progressive web app)
- Features to be tested: Editor, MCQChooser, Autograder, question navigation buttons, missions/quests/paths/etc. listing pages at `academy/*`
- Desktop features should not be affected
- It is recommended to connect your local frontend to the staging backend to test out the missions list page and the missions workspace

### Checklist
